scipy/special/stirling2.h

```diff
@@ -40,7 +43,11 @@ double _stirling2_dp(double n, double k){
         return 0.;
     }
     int arraySize = k <= n - k + 1 ? k : n - k + 1;
-    double *curr = (double *) malloc(arraySize * sizeof(double));
+    auto curr = std::unique_ptr<double[]>{new (std::nothrow) double[arraySize]()};
+    if (curr == nullptr){
```

Comment:
```suggestion
    if (curr == nullptr) {
```
Whitespace nit.

Reply:
I considered it this but it seemed inconsistent with the rest of the file?

Reply:
Actually, the file is not completely consistent. I can't imagine any C/C++ coding standard that would not put a space here. And barring the adoption of some other style guide, my fallback is [PEP 7](https://peps.python.org/pep-0007/), which at least has an example of an `if` statement that has a space before the opening bracket of the code block. (See also [Google's style guide](https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching), which includes the space.)  So let's use the more common convention in new code here.

And if you want, go ahead and fix the other missing spaces (but that's not a blocker).